app/src/main/java/org/simple/clinic/facilitypicker/FacilityPickerView.kt

```diff
@@ -1,47 +1,93 @@
 package org.simple.clinic.facilitypicker
 
+import android.annotation.SuppressLint
 import android.content.Context
 import android.os.Parcelable
 import android.util.AttributeSet
 import android.widget.FrameLayout
+import android.widget.RelativeLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.jakewharton.rxbinding2.support.v7.widget.RxRecyclerView
+import com.jakewharton.rxbinding3.view.detaches
+import com.jakewharton.rxbinding3.widget.textChanges
 import io.reactivex.Observable
+import io.reactivex.rxkotlin.Observables
 import io.reactivex.rxkotlin.cast
+import io.reactivex.rxkotlin.ofType
+import kotlinx.android.synthetic.main.view_facilitypicker.view.*
 import org.simple.clinic.R
 import org.simple.clinic.ReportAnalyticsEvents
 import org.simple.clinic.di.injector
+import org.simple.clinic.facility.Facility
+import org.simple.clinic.facility.change.FacilityListItem
 import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.registration.RegistrationConfig
 import org.simple.clinic.util.unsafeLazy
+import org.simple.clinic.widgets.ItemAdapter
+import org.simple.clinic.widgets.RecyclerViewUserScrollDetector
+import org.simple.clinic.widgets.displayedChildResId
+import org.simple.clinic.widgets.hideKeyboard
 import javax.inject.Inject
 
+private typealias OnFacilitySelected = (Facility) -> Unit
+private typealias OnBackClicked = () -> Unit
+
 class FacilityPickerView(
     context: Context,
     attributeSet: AttributeSet
 ) : FrameLayout(context, attributeSet), FacilityPickerUi, FacilityPickerUiActions {
 
+  @Inject
+  lateinit var effectHandlerFactory: FacilityPickerEffectHandler.Factory
+
+  @Inject
+  lateinit var uiRendererFactory: FacilityPickerUiRenderer.Factory
+
+  @Inject
+  lateinit var config: RegistrationConfig
+
+  var facilitySelectedCallback: OnFacilitySelected? = null
+
+  var backClicked: OnBackClicked? = null
+
+  private val recyclerViewAdapter = ItemAdapter(FacilityListItem.Differ())
+
   init {
     inflate(context, R.layout.view_facilitypicker, this)
     context.injector<Injector>().inject(this)
-  }
 
-  @Inject
-  lateinit var effectHandlerFactory: FacilityPickerEffectHandler.Factory
+    toolbarViewWithSearch.setNavigationOnClickListener { backClicked?.invoke() }
+    toolbarViewWithoutSearch.setNavigationOnClickListener { backClicked?.invoke() }
+
+    facilityRecyclerView.layoutManager = LinearLayoutManager(context)
+    facilityRecyclerView.adapter = recyclerViewAdapter
+
+    searchEditText.requestFocus()
+
+    // Hiding the keyboard without adding a post{} block doesn't seem to work.
+    post { hideKeyboard() }
```

Comment:
Shouldn't we be opening keyboard when this screen is opened since the user can search for a facility?

Reply:
We intentionally do not show it because we expect the location permission to be given so that we can show nearby facilities at the top of the list and the nurse does not need to search most of the time.